core/opendaq/security/include/opendaq/permission_manager.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/baseobject.h>
+#include <coretypes/common.h>
+#include <opendaq/user.h>
+#include <opendaq/permission_config.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_security
+ * @addtogroup opendaq_security_permission_manager PermissionManager
+ * @{
+ */
+
+
+/*!
+ * @brief Enumeration of available access permissions
+ */
+enum Permission
```

Comment:
```suggestion
enum class Permission : uint32_t
```

Reply:
Ommiting class keyword was intentional because of implicit conversion to Int.
For example, when using permission config builder

```
PermissionConfigBuilder().set("admin", Permission::Read | Permission::Write).set("user", Permission::Read);
```

now becomes 

```
PermissionConfigBuilder().set("admin", (Int) Permission::Read | Permission::Write).set("user", (Int) Permission::Read);
```

Reply:
I know, it's just that we're using enum classes everywhere in the SDK, and doing explicit conversions. I think it might be an issue if we're passing them to interfaces, but I'm not 100% sure on that. 

@Unsttopabull Do you think it's okay if we use non-class enumerations here?

Reply:
I have refactored a builder to use a list of Permission values instead.